### PR TITLE
Fail `run_impeller_golden_tests` if `LUCI_CONTEXT && !GOLDCTL`

### PR DESCRIPTION
**Pull request by matanlurey**
Works to prevent future outages like https://github.com/flutter/flutter/issues/147180.

Also added an explicit `--no-skia-gold` flag to use when we don't want Skia Gold used, and used it in the one place we decided (in internal chat) it made sense, `mac_unopt.json` (was added in https://github.com/flutter/engine/commit/67e632882c8b3a5956044352735e082942079e14 for validation-layers testing only).

### CODE REVIEW

**Review by gaaclarke — APPROVED** [review]
lgtm, thanks

**Comment by jonahwilliams** [comment]
https://github.com/flutter/flutter/issues/147180#event-12565727918 landed, rebase this now and we can land it?

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: testing/run_tests.py [hunk 1]
```diff
@@ -1053,7 +1053,7 @@ def __exit__(self, exception_type, exception_value, exception_traceback):
     os.chdir(self.old_cwd)
 
 
-def run_impeller_golden_tests(build_dir: str):
+def run_impeller_golden_tests(build_dir: str, require_skia_gold: bool = False):
```
Comment (gaaclarke) [inline testing/run_tests.py:1056]: We shouldn't need this flag because run_tests.py already has this flag in the form of "--type=impeller-golden", right?

Comment (gaaclarke) [reply testing/run_tests.py:1056]: There's no reason to run the impeller-golden type when require_skia_gold is false?

Comment (matanlurey) [reply testing/run_tests.py:1056]: There is, as-per the PR title - to check if the tests run without validation errors?

Comment (gaaclarke) [reply testing/run_tests.py:1056]: ahh yea, validation errors

**Review by gaaclarke — APPROVED** [review]
lgtm

**Pull request merged**